lib/has_friendship/friendable.rb

```diff
@@ -44,33 +50,35 @@ def friend_request(friend)
       end
 
       def accept_request(friend)
-        transaction do
-          pending_friendship = HasFriendship::Friendship.find_friendship(friend, self)
-          pending_friendship.status = 'accepted'
-          pending_friendship.save
-
-          requeseted_friendship = HasFriendship::Friendship.find_friendship(self, friend)
-          requeseted_friendship.status = 'accepted'
-          requeseted_friendship.save
+        friend_engine(friend) do |one, other|
+           HasFriendship::Friendship.find_friendship(one, other).update(status: 'accepted' )
         end
       end
 
       def decline_request(friend)
-        transaction do
-          HasFriendship::Friendship.find_friendship(friend, self).destroy
-          HasFriendship::Friendship.find_friendship(self, friend).destroy
+        friend_engine(friend) do |one, other|
+          HasFriendship::Friendship.find_friendship(one, other).destroy
+        end
+      end
+
+      alias_method :remove_friend, :decline_request
+
+      def block_friend(friend)
+        friend_engine(friend) do |one, other|
+          HasFriendship::Friendship.find_friendship(one, other).update(status: 'blocked' )
         end
       end
 
-      def remove_friend(friend)
+
+      def friend_engine(friend)
         transaction do
-          HasFriendship::Friendship.find_friendship(friend, self).destroy
-          HasFriendship::Friendship.find_friendship(self, friend).destroy
+          yield(self, friend)
+          yield(friend, self)
         end
       end
 
       def friends_with?(friend)
-        HasFriendship::Friendship.where(friend_id: friend.id, status: 'accepted').present?
+        HasFriendship::Friendship.where(friendable_id: self.id, friend_id: friend.id, status: 'accepted').present?
```

Comment:
This was fixed in 0.0.3 but I forgot to merge it to `master`. I just merged it. Can you pull from my `master`, and push again?